lyncs_quda/gauge_field.py

```diff
@@ -147,7 +150,11 @@ def ncol(self):
     @property
     def order(self):
         "Data order of the field"
-        return "FLOAT2"
+        dofs = self.dofs_per_link
+        if dofs == 8 or dofs == 12:
+            return "FLOAT4"
```

Comment:
modified.  For the older version of QUDA, #ifdef FLOAT8, QUDA_FLOAT8_GAUGE_ORDER should be returned in case of reconstruct == 8 and prec == (half, quarter), and FLOAT8 is used to be defined at least on our machine via cmake.

In the new version, probably because float8 and short8 are no longer built-in vector type in CUDA, this if-clause is removed, and then you are right.  If not double, even when reconstruct==8, QUDA_FLOAT4_GAUGE_ORDER is returned (c.f. gauge_field.cpp).

So I basically make my code consistent with the new version of QUDA.